### PR TITLE
Adjust to new names and add better documentations and e2e

### PR DESCRIPTION
**Pull request by anthonyvercolano**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-node/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: provisioning/service/src/interfaces.ts [hunk 1]
```diff
@@ -18,138 +30,336 @@ export type AttestationTypes = 'none' | 'tpm' | 'x509';
...
+ * This object is not provided by the application.  It is populated by the service from
+ * a provided certificate.
  */
 export interface X509CertificateInfo {
+  /**
+   * For clientCertificates the subjectName is used as the deviceId on the IoT Hub.
+   */
   subjectName: string;
+  /**
+   * A hash of the certificate whose size is 160 bits.
```
Comment (BertKleewein) [inline provisioning/service/src/interfaces.ts]: >A hash of the certificate whose size is 160 bits [](start = 5, length = 48)

awkward.  It sounds like the certificate is 160 bits.  Suggest "a 160 bit hash of the certificate"?  Same for sha256

Comment (anthonyvercolano) [reply provisioning/service/src/interfaces.ts]: Done.

---
In reply to: [149230292](https://github.com/Azure/azure-iot-sdk-node/pull/137#discussion_r149230292) [](ancestors = 149230292)

File: provisioning/e2e/_service_create_delete.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+'use strict';
+
+var assert = require('chai').assert;
+var debug = require('debug')('azure-device-provisioning-e2e');
+var uuid = require('uuid');
+
+
+var provisioningServiceClient = require('azure-iot-provisioning-service').ProvisioningServiceClient;
+
+var serviceClient = provisioningServiceClient.fromConnectionString(process.env.IOTHUB_PROVISIONING_SERVICE_CONNECTION_STRING);
```
Comment (pierreca) [inline provisioning/e2e/_service_create_delete.js]: >IOTHUB [](start = 79, length = 6)

nit: IOT not IOTHUB

File: provisioning/service/src/interfaces.ts [hunk 3]
```diff
@@ -18,138 +30,336 @@ export type AttestationTypes = 'none' | 'tpm' | 'x509';
...
 
 /**
  * X509 certificate info.
+ *
+ * This object is not provided by the application.  It is populated by the service from
+ * a provided certificate.
  */
 export interface X509CertificateInfo {
+  /**
+   * For clientCertificates the subjectName is used as the deviceId on the IoT Hub.
```
Comment (pierreca) [inline provisioning/service/src/interfaces.ts]: >clientCertificates [](start = 9, length = 18)

camelcasing suggests a property of something?

Comment (pierreca) [inline provisioning/service/src/interfaces.ts]: >subjectName [](start = 32, length = 11)

same here - also I think we should refer to this as common name ? or CN property of the subject?

**Pull request merged**